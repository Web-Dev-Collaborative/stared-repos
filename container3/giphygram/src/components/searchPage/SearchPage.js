import React from 'react';
import SearchResultsContainerConnected from '../../containers/searchResultsContainer/SearchResultsContainer';
import SearchSummaryContainerConnected from '../../containers/searchSummaryContainer/SearchSummaryContainer';
import SearchViewTabsContainerConnected from '../../containers/searchViewTabsContainer/SearchViewTabsContainer';

export class SearchPage extends React.Component {
  render() {
    return (
      <React.Fragment>
        <div className="row">
          <div className="col-6">
            <SearchSummaryContainerConnected />
          </div>
          <div className="col-6 text-right">
            <SearchViewTabsContainerConnected />
          </div>
        </div>
        <SearchResultsContainerConnected />
      </React.Fragment>
    );
  }
}
