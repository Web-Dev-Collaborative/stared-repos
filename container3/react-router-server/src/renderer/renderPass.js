import React from 'react';
import { renderToString, renderToStaticMarkup } from 'react-dom/server';
import AsyncRenderer from '../components/AsyncRenderer';
import removeDuplicateModules from '../utils/removeDuplicateModules';

const renderPass = (context, element, staticMarkup = false) => {
  context.callback = () => {
    if (context.finishedLoadingModules && !context.statesRenderPass) {
      context.statesRenderPass = true;
      context.renderResult = renderPass(context, element, staticMarkup);
      if (context.fetchingStates <= 0 && context.modulesLoading <= 0) {
        context.resolve({
          html: context.renderResult,
          state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
          modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
        });
      }
    } else if (context.finishedLoadingModules && context.statesRenderPass || !context.hasModules) {
      context.renderResult = renderPass(context, element, staticMarkup);
      if (context.fetchingStates <= 0 && context.modulesLoading <= 0) {
        context.resolve({
          html: context.renderResult,
          state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
          modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
        });
      }
    }
  };

  let component = (
    <AsyncRenderer context={context}>
      {element}
    </AsyncRenderer>
  );

  let result;
  try {
    if (staticMarkup) {
      result = renderToStaticMarkup(component);
    } else {
      result = renderToString(component);
    }
  } catch (e) {
    return context.reject(e);
  }

  if (!context.hasModules && !context.hasStates) {
    context.resolve({
      html: result,
      state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
      modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
    });
  }
  return result;
};

export default renderPass;
