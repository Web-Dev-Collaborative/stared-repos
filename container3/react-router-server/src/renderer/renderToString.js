import renderPass from './renderPass';
import { clear } from '../module/cache';

export default (element) => {
  return new Promise((resolve, reject) => {
    clear();
    const context = {
      resolve, reject,
      modulesLoading: 0,
      fetchingStates: 0
    };
    renderPass(context, element);
  });
};
