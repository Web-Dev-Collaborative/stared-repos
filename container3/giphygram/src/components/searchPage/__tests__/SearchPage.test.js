import React from 'react';
import renderer from 'react-test-renderer';
import { SearchPage } from '../SearchPage';

jest.mock(
  '../../../containers/searchResultsContainer/SearchResultsContainer',
  () => 'SearchResultsContainer',
);

jest.mock(
  '../../../containers/searchSummaryContainer/SearchSummaryContainer',
  () => 'SearchSummaryContainer',
);

jest.mock(
  '../../../containers/searchViewTabsContainer/SearchViewTabsContainer',
  () => 'SearchViewTabsContainer',
);

describe('SearchPage', () => {
  it('should be rendered correctly', () => {
    const tree = renderer
      .create(<SearchPage />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
