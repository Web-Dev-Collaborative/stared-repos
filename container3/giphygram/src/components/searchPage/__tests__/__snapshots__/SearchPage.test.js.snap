// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage should be rendered correctly 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <SearchSummaryContainer />
    </div>
    <div
      className="col-6 text-right"
    >
      <SearchViewTabsContainer />
    </div>
  </div>,
  <SearchResultsContainer />,
]
`;
